solc/CommandLineInterface.cpp

```diff
@@ -877,11 +877,15 @@ void CommandLineInterface::handleCombinedJSON()
 	if (m_options.compiler.combinedJsonRequests->ast)
 	{
 		output[g_strSources] = Json::Value(Json::objectValue);
+		unsigned sourceIndex = 0;
 		for (auto const& sourceCode: m_fileReader.sourceUnits())
 		{
-			ASTJsonExporter converter(m_compiler->state(), m_compiler->sourceIndices());
 			output[g_strSources][sourceCode.first] = Json::Value(Json::objectValue);
-			output[g_strSources][sourceCode.first]["AST"] = converter.toJson(m_compiler->ast(sourceCode.first));
+			output[g_strSources][sourceCode.first]["AST"] = ASTJsonExporter(
+																m_compiler->state(),
+																m_compiler->sourceIndices()
+															).toJson(m_compiler->ast(sourceCode.first));
+			output[g_strSources][sourceCode.first]["id"] = sourceIndex++;
```

Comment:
Actually, looking at this a bit more closely: this should really be
```suggestion
			output[g_strSources][sourceCode.first]["id"] = m_compiler->sourceIndices().at(sourceCode.first);
```

Looking at the comments, ``sourceIndices()`` gives you exactly the ordering of `sourceNames()`` in the compiler stack, which is what ``StandardCompiler`` iterates over - and which may *not* be the file reader order here, right?